java/common/common-task/src/test/java/sleeper/task/common/RunCompactionTasksTest.java

```diff
@@ -153,6 +153,45 @@ void shouldCreateTasksToMaxWhenJobsOnQueuePlusExistingTasksGreaterThanMax() {
             assertThat(scaleToHostsRequests).containsExactly(3);
             assertThat(launchTasksRequests).containsExactly(1);
         }
+
+        @Test
+        void shouldCreateNoTasksAndScaleWhenExistingTasksPresentPlusExistingTasksLessThanMax() {
+            // Given
+            instanceProperties.setNumber(MAXIMUM_CONCURRENT_COMPACTION_TASKS, 5);
+
+            // When
+            runTasks(noJobsOnQueue(), existingTasks(3));
+
+            // Then
+            assertThat(scaleToHostsRequests).containsExactly(3);
+            assertThat(launchTasksRequests).isEmpty();
+        }
+
+        @Test
+        void shouldCreateNoTasksAndScaleWhenExistingTasksPresentPlusExistingTasksEqualToMax() {
+            // Given
+            instanceProperties.setNumber(MAXIMUM_CONCURRENT_COMPACTION_TASKS, 3);
+
+            // When
+            runTasks(noJobsOnQueue(), existingTasks(3));
+
+            // Then
+            assertThat(scaleToHostsRequests).containsExactly(3);
+            assertThat(launchTasksRequests).isEmpty();
+        }
+
+        @Test
+        void shouldCreateNoTasksAndScaleWhenExistingTasksPresentPlusExistingTasksGreaterThanMax() {
```

Comment:
This could be something like "shouldScaleHostsWhenNoJobsOnQueueAndExistingTasksGreaterThanMax"?